lua/wire/wireshared.lua

```diff
@@ -369,6 +369,45 @@ function WireLib.ErrorNoHalt(message)
 	end
 end
 
+--- Generate a random version 4 UUID and return it as a string.
+function WireLib.GenerateUUID()
+	-- It would be easier to generate this by word rather than by byte, but
+	-- MSVC's RAND_MAX = 0x7FFF, which means math.random(0, 0xFFFF) won't
+	-- return all possible values.
+	local bytes = {}
+	for i = 1, 16 do bytes[i] = math.random(0, 0xFF) end
+	bytes[7] = bit.bor(0x40, bit.band(bytes[7], 0x0F))
+	bytes[9] = bit.bor(0x80, bit.band(bytes[7], 0x3F))
+	return string.format("%02x%02x%02x%02x-%02x%02x-%02x%02x-%02x%02x-%02x%02x%02x%02x%02x%02x", unpack(bytes))
+end
+
+local UUID_MESSAGE_NAME = "WireLib.GetServerUUID"
```

Comment:
Renamed to `PERSISTENT_UUID_KEY` which better reflects that it's used in multiple ways.